packages/apps/human-app/frontend/src/shared/providers/notifications-provider.tsx

```diff
@@ -0,0 +1,52 @@
+import { type ReactNode } from 'react';
+import { SnackbarProvider } from 'notistack';
+import CheckCircleIcon from '@mui/icons-material/CheckCircle';
+import ErrorIcon from '@mui/icons-material/Error';
+import { type TopNotificationType } from '@/shared/hooks/use-notification';
+import { colorPalette as lightColorPalette } from '@/shared/styles/color-palette';
+import { exhaustiveMatchingGuard } from '@/shared/helpers/exhaustive-matching-guard';
+
+interface NotificationProviderProps {
+  children: ReactNode;
+  maxSnacks?: number;
+}
+
+const MAX_NOTIFICATIONS_VISIBLE = 5;
+
+const getNotificationIconByType = (
+  type: TopNotificationType,
+  sx?: Record<string, string>
+) => {
+  switch (type) {
+    case 'success':
+      return <CheckCircleIcon sx={{ fill: lightColorPalette.white, ...sx }} />;
+
+    case 'warning':
+      return <ErrorIcon sx={{ fill: lightColorPalette.white, ...sx }} />;
+
+    default: {
+      exhaustiveMatchingGuard(type);
```

Comment:
You pass `type` here, but don't use it in `exhaustiveMatchingGuard`, so I suppose you wanted to have some informative error thrown :)

I suppose in this case it should be fine to not have some abstract helper, but just throw error in place, like
```
default:
  throw new Error(`Unexpected notification icon type requested: ${type}`);
```